### ISSUE TITLE
Suggestion: new identification method

### ISSUE THREAD

**Issue by nishihatapalmer** [digital-preservation/droid#340]
File formats are currently identified by three methods:

 * Extension
 * Signature
 * Container

The method of identification reflects the method used to identify a file, but it also represents a degree of certainty about the identification.  Extensions are not particularly reliable, signatures are reasonably reliable, and containers are the most reliable.  

We believe containers are the most reliable because not only has the base container format been identified with a signature, it is then parsed as an instance of that type of file, and sub-files are examined with further binary signatures applied in many cases to the sub-files inside the container.

I propose there should be a new identification method: Archive.

When we identify an archive file format via a binary signature, and then go on to successfully parse it, identifying the files inside it, we have a stronger identification than just the base archive format identification achieved with a binary signature.  Code can actually parse this type of file, which gives much stronger credence that it really is an instance of this file.

Therefore, archive files identified by a binary signature, which are then successfully parsed as an instance of that file type, should have a stronger identification method than those which are merely identified by a binary signature.

This would give:

 * Extension  - identified just on file extension.
 * Signature  - identified by binary signature
 * Archive - identified by binary signature and parsing the files inside the archive file format.
 * Container - identified by binary signature, parsing the files inside the container, and identifying binary signatures inside them.

**Comment by nishihatapalmer** [comment]
Of course, adding a new identification method has a good chance of screwing up all sorts of reports people may currently use.  

Any new identification method would probably have to be a configurable extra (with the default being existing behaviour).

**Comment by steve-daly** [comment]
Too much of a breaking change to consider currently.
